File: fixtures/scrapy_output/multipart_post.py
from scrapy import FormRequest, Spider


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        # To implement file uploads in Scrapy, see:
        # https://stackoverflow.com/a/39312565
        formdata = {
            'attributes': '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}',
        }
        headers = {
            'Authorization': 'Bearer ACCESS_TOKEN',
        }
        yield FormRequest(
            'https://upload.box.com/api/2.0/files/content',
            formdata=formdata,
            callback=self.parse,
            headers=headers,
        )

    def parse(self, response):
        pass

File: test/fixtures/javascript-xhr/multipart_post.js
import * as fs from 'fs';

const form = new FormData();
form.append('attributes', '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}');
form.append('file', fs.readFileSync('myfile.jpg'), 'myfile.jpg');

let xhr = new XMLHttpRequest();
xhr.withCredentials = true;
xhr.open('POST', 'http://localhost:28139/api/2.0/files/content');
xhr.setRequestHeader('Authorization', 'Bearer ACCESS_TOKEN');

xhr.onload = function() {
  console.log(xhr.response);
};

xhr.send(form);
